jib-core/src/test/java/com/google/cloud/tools/jib/image/ReproducibleLayerBuilderTest.java

```diff
@@ -183,6 +183,84 @@ public void testToBlob_reproducibility() throws IOException {
   }
 
+  @Test
+  public void testBuild_parentDirBehavior() throws IOException {
+    Path testRoot = temporaryFolder.getRoot().toPath();
+
+    // the path doesn't really matter on source files, but these are structured
+    Path parent = Files.createDirectories(testRoot.resolve("aaa"));
+    Path fileA = Files.createFile(parent.resolve("fileA"));
+    Path ignoredParent = Files.createDirectories(testRoot.resolve("bbb-ignored"));
+    Path fileB = Files.createFile(ignoredParent.resolve("fileB"));
+    Path fileC =
+        Files.createFile(Files.createDirectories(testRoot.resolve("ccc-absent")).resolve("fileC"));
+
+    Blob layer =
+        new ReproducibleLayerBuilder(
+                ImmutableList.of(
+                    new LayerEntry(
+                        parent,
+                        AbsoluteUnixPath.get("/root/aaa"),
+                        FilePermissions.fromOctalString("111"),
+                        Instant.ofEpochSecond(10)),
+                    new LayerEntry(
+                        fileA,
+                        AbsoluteUnixPath.get("/root/aaa/fileA"),
+                        FilePermissions.fromOctalString("222"),
+                        Instant.ofEpochSecond(20)),
+                    new LayerEntry(
+                        fileB,
+                        AbsoluteUnixPath.get("/root/bbb-ignored/fileB"),
+                        FilePermissions.fromOctalString("333"),
+                        Instant.ofEpochSecond(30)),
+                    new LayerEntry(
+                        ignoredParent,
+                        AbsoluteUnixPath.get("/root/bbb-ignored"),
+                        FilePermissions.fromOctalString("444"),
+                        Instant.ofEpochSecond(40)),
+                    new LayerEntry(
+                        fileC,
+                        AbsoluteUnixPath.get("/root/ccc-absent/file3"),
+                        FilePermissions.fromOctalString("555"),
+                        Instant.ofEpochSecond(50))))
+            .build();
+
+    Path tarFile = temporaryFolder.newFile().toPath();
+    try (OutputStream out = new BufferedOutputStream(Files.newOutputStream(tarFile))) {
+      layer.writeTo(out);
+    }
+
+    try (TarArchiveInputStream in = new TarArchiveInputStream(Files.newInputStream(tarFile))) {
+      // root (default folder permissions)
+      TarArchiveEntry root = in.getNextTarEntry();
+      Assert.assertEquals(040755, root.getMode());
+      Assert.assertEquals(Instant.ofEpochSecond(1), root.getModTime().toInstant());
+
+      // parentAAA (custom permissions, custom timestamp)
+      TarArchiveEntry rootParentAAA = in.getNextTarEntry();
+      Assert.assertEquals(040111, rootParentAAA.getMode());
+      Assert.assertEquals(Instant.ofEpochSecond(10), rootParentAAA.getModTime().toInstant());
+
+      // skip over fileA
+      in.getNextTarEntry();
+
+      // parentBBB (default permissions - ignored custom permissions, since fileB added first)
```

Comment:
Add a TODO that this should really be fixed and return `040444` and `Instant.ofEpochSecond(40)`?